src/modules/dataman/dataman.cpp

```diff
@@ -1098,13 +1092,16 @@ dm_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void
 	work->write_params.count = count;
 
 	/* Enqueue the item on the work queue and wait for the worker thread to complete processing it */
-	return (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	ssize_t ret = (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	perf_end(_dm_write_perf);
```

Comment:
There are more return's in between, where perf_end is not called (same below).

Reply:
Fixed.